src/project/network/ConsumptionMacro.java

```diff
@@ -0,0 +1,58 @@
+package project.network;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+
+public class ConsumptionMacro {
+
+	private ConsumptionType consumpType;
+	private static int Cursor;

```

Comment:
Pourquoi static ?

Reply:
C'est un relent de la version précédente, en soi c'est toujours une sécurité même si il ne devrait y avoir qu'un seule ConsumptionMacro

Reply:
Mais du coup tu ne peux pas instancier plusieurs ConsumptionMacro sans qu'ils soient synchronisés. Autant avoir toute ta classe en statique alors si l'instancier est inutile.

Reply:
Et si tu le laisses statique tu dois y accéder via la classe, c'est-à-dire faire `ConsumptionMacro.Cursor` plutôt que `this.Cursor`.java